rskj-core/src/main/java/co/rsk/pcc/altBN128/AltBN128.java

```diff
@@ -4,15 +4,20 @@
 import co.rsk.pcc.altBN128.impls.AbstractAltBN128;
 import co.rsk.pcc.altBN128.impls.GoAltBN128;
 import co.rsk.pcc.altBN128.impls.JavaAltBN128;
+import co.rsk.pcc.altBN128.impls.JavaRSKIP197AltBN128;
 
 public class AltBN128 extends AbstractAltBN128 {
     private AbstractAltBN128 implementation;
 
-    public AltBN128() {
+    public AltBN128(boolean rskip197active) {
         if (Utils.isLinux()) {
             this.implementation = new GoAltBN128();
```

Comment:
GoAltBN128 behaves like both RSKIP197 and the default implementaitons? Because I smell a hardfork here

Reply:
Go behaves as Java with rskip197. Before rskip197 they (java & go) behave diff but ended up with the same result thanks to the logic in the execution in each Precompiled.
Now that we want to throw an error, the expected result is diff, thats why we have to fork with the implementation of JavaRSKIP197.
Is that what you mean? Or am I missing sthg?

Reply:
Also take a look at BN128PrecompiledContract.execute()

Reply:
Instead of using a flag you can create another constructor with `javaImpl` as parameter (check the other comment I've made) and mark `Altbn()` as deprecated